resample/tests/test_bootstrap.py

```diff
@@ -0,0 +1,224 @@
+import numpy as np
+from numpy.testing import assert_equal, assert_almost_equal
+import pytest
+from scipy import stats
+
+from resample.bootstrap import (
+    resample,
+    bootstrap,
+    confidence_interval,
+    _fit_parametric_family,
+)
+
+PARAMETRIC_CONTINUOUS = {
+    # use scipy.stats names here
+    "norm",
+    "t",
+    "laplace",
+    "logistic",
+    "f",
+    "beta",
+    "gamma",
+    "lognorm",
+    "invgauss",
+    "pareto",
+}
+PARAMETRIC_DISCRETE = {"poisson"}
+PARAMETRIC = PARAMETRIC_CONTINUOUS | PARAMETRIC_DISCRETE
+NON_PARAMETRIC = {"ordinary", "balanced"}
+ALL_METHODS = NON_PARAMETRIC | PARAMETRIC
+
+
+@pytest.fixture
+def rng():
+    return np.random.Generator(np.random.PCG64(1))
+
+
+@pytest.mark.parametrize("method", ALL_METHODS)
+def test_resample_shape_1d(method):
+    if method == "beta":
+        x = (0.1, 0.2, 0.3)
+    else:
+        x = (1, 2, 3)
+    n_rep = 5
+    count = 0
+    with np.errstate(invalid="ignore"):
+        for bx in resample(x, n_rep, method=method):
+            assert len(bx) == len(x)
+            count += 1
+    assert count == n_rep
+
+
+@pytest.mark.parametrize("method", NON_PARAMETRIC | {"norm"})
+def test_resample_shape_2d(method):
+    x = [(1, 2), (4, 3), (6, 5)]
+    n_rep = 5
+    count = 0
+    for bx in resample(x, n_rep, method=method):
+        assert bx.shape == np.shape(x)
+        count += 1
+    assert count == n_rep
+
+
+@pytest.mark.parametrize("method", NON_PARAMETRIC)
+def test_resample_shape_4d(method):
+    x = np.ones((2, 3, 4, 5))
+    n_rep = 5
+    count = 0
+    for bx in resample(x, 5, method=method):
+        assert bx.shape == np.shape(x)
+        count += 1
+    assert count == n_rep
+
+
+@pytest.mark.parametrize("method", NON_PARAMETRIC | PARAMETRIC_CONTINUOUS)
+def test_resample_1d_parametric(method, rng):
```

Comment:
Is there any way to simplify this test? Also we may want to rename since it's not only for parametric sampling.

Reply:
I always aim to write the most compact code to achieve some goal, so if there is a simpler way, you need to suggest it.

This is a statistical test which checks that the resampling actually produces distributions which agree with the original distribution. To compare the realized with the expected distributions, I use a chi-square test. I then check whether the pvalues are flat. I found that this test is very sensitive to any statistical deviations.

I am very satisfied with the result and I spend a lot of time on this. This is the test to convince people that this module works correctly, so it is rather important.

Reply:
I fixed the name.

Reply:
I guess it's fine as long as we know it'll fail randomly and that's documented

Reply:
I made sure it cannot fail randomly by fixing the seed and the RNG. As you can see, the test passes on all machines and all Python versions.